server/conditions/conditions.go

```diff
@@ -80,11 +89,24 @@ func applyCardsToState(cards []cards.Card, state *pogo.GameState, conditions []C
 	// loop through each action group and check it against every condition
 	for _, actionGroup := range actionGroups {
 		//fmt.Printf("processing action group with %v actions\n", len(actionGroup))
-		for _, cond := range conditions {
-			// this is the step that actually checks each condition
-			var condErr = cond.Apply(state, actionGroup)
-			if condErr != nil {
-				return state, condErr
+		var dirty = true
+		var loops = 0
+		for dirty {
+			dirty = false
+			for _, cond := range conditions {
+				// this is the step that actually checks each condition
+				var steps, condErr = cond.Apply(state, actionGroup, 0)
+				if condErr != nil {
+					return state, condErr
+				}
+				if steps != 0 {
+					dirty = true
+					loops++
+					break
+				}
+			}
+			if loops > MaxConditionSteps {
+				return state, fmt.Errorf("failed to apply conditions, reached maximum loop counter of %v", MaxConditionSteps)
```

Comment:
This loop now uses the same sort of logic contained within the conditions themselves.  If a condition is applied and it marks the state as "dirty", not only will it run internally until that state is clean, but it will also run the entire list of conditions again until all of the conditions say that the state is clean.  This is the fix that actually fixed #65.